subsys/settings/src/settings_file.c

```diff
@@ -81,23 +84,65 @@ static int settings_file_load_priv(struct settings_store *cs, line_load_cb cb,
 	}
 
 	while (1) {
+		bool is_duplicate = false;
```

Comment:
You could easily use the exact same methodology in fcb, doing the same thing makes it easier to maintain.

Reply:
I would rather pass the same variable name into file - is_duplicate does not explain everything and I am worried that it might be used for additional checking in the future. pass_entry explains what this variable does.

Reply:
Done

Reply:
I agree, I would not use is_duplicate, but use a boolean load = true. This explains everything, you set load to false when not needed. The methodology in the file is a lot simpler than in the fcb backend, so keep this methodology.

Reply:
I do not think so. Moreover - to much of tabs is required in files and it makes it hard to keep line length limit. Splitting it into two functions line in fcb backend.

Reply:
It is very easy to put this in a function for the file as well. This is no reason. Keep the methods equal! Just have this function return 0 if it finds a match, and an error code when it doesn't (a error: end of file or whatever error means loading is needed).

Reply:
I see no reason to flatten the function return value so much. It might make sense to react in the same way when an error occur and no match is found. The functionality is the same now in both backends.

Reply:
@rakons, I do see a reason for both backends to flatten out the function return. At the moment settings reads as much as it can, when you don't flatten out the error settings will return no value at all because there is a error later in the backend, even when this error has nothing to do with the value being read.

Reply:
Did you look into the current code?